packages/@react-stately/data/src/useTreeData.ts

```diff
@@ -227,16 +238,19 @@ export function useTreeData<T extends object>(options: TreeOptions<T>): TreeData
       return map.get(key);
     },
     insert(parentKey: Key | null, index: number, ...values: T[]) {
-      setItems(items => {
-        let nodes = buildTree(values, parentKey);
+      setItems(({items, mappy: originalMap}) => {
+        let {items: nodes, mappy: newMap} = buildTree(values, parentKey, originalMap);
```

Comment:
won't this still result in originalMap being mutated? `buildTree` only creates a new map if `acc` is not provided, otherwise it reuses it.

Reply:
not quite sure I follow here. We are passing originalMap here to buildTree to build the newMap for the `parentKey === null` case. If we didn't create pass it, buildTree would create a new map that would then only have the new nodes that are being inserted here instead of the full modified tree map since `values` is just the new values being inserted

modifying the originalMap here shouldn't matter right since we are now storing the map in state and accessing the previous map inside the setState call?

Reply:
yeah here https://github.com/adobe/react-spectrum/pull/5835/files#diff-c1a25c31946ec151047ebc265b0f5cf0971d990a3407c0dd66b7beecb1be4b24R151 we mutate the map that gets passed into `buildTree`, thereby mutating the map stored in state.